artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -270,6 +271,15 @@
 
    private final QueueFactory factory;
 
+   private final AtomicBoolean dispatching = new AtomicBoolean();
```

Comment:
The logic that involves using dispatching and dispatchStartTime could be simplyfied using just dispatchStartTime? 
I would use and AtomicLongFieldUpdater for that in case

Reply:
I wanted that when dispatching the quickest eval is done, thus why the seperate boolean. As it is hot path. Agree could just use != -1 check, memory vs cpu cycle. Let me know your thought 

Agree re atomic updater, i can make this change.

Reply:
>  Agree could just use != -1 check, memory vs cpu cycle

TBH to me are good enough both, but if we can just check !=-1 it would be welcome, even if less readable...
I'm more worried about the writes cost then the reads :P

Reply:
@franz1981 so after going back to see if i could just use != -1, i remembered why i couldnt it would mean adding a System.currentMillis call on the hot path, which is very costly.

I have though moved to using AtomicUpdaters , though :)